src/test/java/com/atlassian/bitbucket/jenkins/internal/applink/oauth/serviceprovider/rest/OauthCrumbExclusionTest.java

```diff
@@ -0,0 +1,63 @@
+package com.atlassian.bitbucket.jenkins.internal.applink.oauth.serviceprovider.rest;
+
+import com.atlassian.bitbucket.jenkins.internal.applink.oauth.serviceprovider.OAuthRequestUtils;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class OauthCrumbExclusionTest {
+
+    @Spy
+    private FilterChain chain;
+    private OauthCrumbExclusion crumbExclusion;
```

Comment:
nit, but the ordering here is a bit out of whack. I say put your mocks first, then your spies and lastly the thing that are proper instances.

Reply:
Thanks!